%
% FILE NAME:    masterScriptMouseGEM.m
%
%
% PURPOSE: This script is for reconstruction of the Mouse-GEM, by using
%          the Human-GEM as template and taking into account mouse-specific
%          pathways/reactions.
%
%          Annotation files in tsv format are also generated by
%          integrating human and mouse-specific annotation information.
%


%% Add path
addpath(genpath('../../Mouse-GEM/'));


%% Prepare Mouse ortholog pairs and species-specific network

% get ortholog pairs from human to mouse
mouseOrthologPairs = extractAllianceGenomeOrthologs('human2MouseOrthologs.tsv');

% load species-specific rxns and mets
rxnsToAdd = importTsvFile('mouseSpecificRxns.tsv');
metsToAdd = importTsvFile('mouseSpecificMets.tsv');


%% Generate Mouse-GEM
[mouseGEM, speciesSpecNetwork, gapfillNetwork]=updateAnimalGEM(...
    mouseOrthologPairs,rxnsToAdd,metsToAdd,'Mouse-GEM');


%% Update annotation files
[rxnAssoc, metAssoc] = updateAnimalAnnotations(mouseGEM,rxnsToAdd,metsToAdd);


%% Save annotation into tsv files, and the model into mat, yml, and xml

% sanity check
if isequal(rxnAssoc.rxns, mouseGEM.rxns) && isequal(metAssoc.mets, mouseGEM.mets)
	fprintf('sanity check passed.\n');
    exportTsvFile(rxnAssoc,'../model/reactions.tsv');
    exportTsvFile(metAssoc,'../model/metabolites.tsv');
end

mouseGEM.geneShortNames = mouseGEM.genes;
save('../model/Mouse-GEM.mat', 'mouseGEM');
exportYaml(mouseGEM, '../model/Mouse-GEM.yml');
mouseGEM = annotateGEM(mouseGEM,'../model',{'rxn','met'});  % add annotation data to structure
mouseGEM.id = regexprep(mouseGEM.id,'-','');  % remove dash from model ID since it causes problems with SBML I/O
exportModel(mouseGEM, '../model/Mouse-GEM.xml');

